### PR TITLE
Add react node text util

### PR DESCRIPTION
**Pull request by gdostie**
### Description

I wanted to implement a simple matchFilter for the SingleSelectWithFilter that can match on the `content` attribute, which is a ReactNode.

### Proposed Changes

- I added a small util called `getReactNodeTextContent` that converts a ReactNode into a string by keeping only the text content (it strips the html tags)

### Potential Breaking Changes

None.

### Acceptance Criteria

- [x] The proposed changes are covered by unit tests
- [x] The potential breaking changes are clearly identified
- [x] [README.md](https://github.com/coveo/react-vapor/blob/master/README.md) is adjusted to reflect the proposed changes (if relevant)

### CODE REVIEW

File: packages/react-vapor/src/utils/JSXUtils.tsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import * as React from 'react';
+import {renderToStaticMarkup} from 'react-dom/server';
+import * as s from 'underscore.string';
+
+export type JSXRenderable = JSX.Element | JSX.Element[] | string | number;
+
+export const getReactNodeTextContent = (node: React.ReactNode): string => {
+    const HTMLString = renderToStaticMarkup(<div>{node}</div>);
+    return s.stripTags(HTMLString).replace(/\s{2,}/g, ' ');
```
Comment (gdostie) [inline packages/react-vapor/src/utils/JSXUtils.tsx]: The replace here simply converts whitespaces chunks of 2 occurences or more into a single whitespace.
```
'Pizza is  my favorite   meal.' --> 'Pizza is my favorite meal.'
```

File: packages/react-vapor/src/utils/JSXUtils.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+describe('JSXUtils', () => {
+    describe('getReactNodeTextContent', () => {
+        it('should return an empty string when receiving falsy values as react node', () => {
+            expect(getReactNodeTextContent(null)).toBe('');
+            expect(getReactNodeTextContent(undefined)).toBe('');
+            expect(getReactNodeTextContent(<></>)).toBe('');
+        });
+
+        it('should return the text content of the react node reveived as argument', () => {
+            expect(getReactNodeTextContent(
```
Comment (nanani-coveo) [inline packages/react-vapor/src/utils/JSXUtils.spec.tsx]: ** received

File: packages/react-vapor/src/utils/JSXUtils.tsx (hunk 1 above)
Comment (GermainBergeron) [reply packages/react-vapor/src/utils/JSXUtils.tsx]: I'd replace with `s(HTMLString).stripTags().clean().value();` I think

Comment (gdostie) [reply packages/react-vapor/src/utils/JSXUtils.tsx]: The typings don't support it, dunno why. It works at runtime though.
image

**Pull request merged**